engine/crates/engine-v2/schema/src/builder/mod.rs

```diff
@@ -88,23 +79,75 @@ impl SchemaBuilder {
                 urls: Vec::new(),
                 input_values: SchemaInputValues::default(),
             },
+        };
+
+        for (idx, input_value_definition) in take(&mut config.graph.input_value_definitions).into_iter().enumerate() {
+            if is_inaccessible(&config.graph, input_value_definition.directives) {
+                builder.id_mapper.skip(federated_graph::InputValueDefinitionId(idx))
+            } else {
+                builder
+                    .schema
+                    .input_value_definitions
+                    .push(input_value_definition.into());
+            }
         }
+
+        for (idx, enum_value) in take(&mut config.graph.enum_values).into_iter().enumerate() {
+            if is_inaccessible(&config.graph, enum_value.composed_directives) {
+                builder.id_mapper.skip(federated_graph::EnumValueId(idx))
+            } else {
+                builder.schema.enum_values.push(enum_value.into());
+            }
+        }
+
+        for (i, field) in config.graph.fields.iter().enumerate() {
+            if is_inaccessible(&config.graph, field.composed_directives) {
+                builder.id_mapper.skip(federated_graph::FieldId(i));
+            }
+        }
+
+        builder.schema.input_objects = take(&mut config.graph.input_objects)
+            .into_iter()
+            .map(|input_object| builder.convert_input_object(input_object))
+            .collect();
+
+        builder.schema.enums = take(&mut config.graph.enums)
+            .into_iter()
+            .map(|enm| builder.convert_enum(enm))
+            .collect();
+
+        builder.schema.unions = take(&mut config.graph.unions)
+            .into_iter()
+            .map(|union| Union {
+                name: union.name.into(),
+                description: None,
+                possible_types: union
+                    .members
+                    .into_iter()
+                    .filter(|object_id| !is_inaccessible(&config.graph, config.graph[*object_id].composed_directives))
+                    .map(Into::into)
+                    .collect(),
+                composed_directives: union.composed_directives.into(),
+            })
+            .collect();
+
+        builder
     }
 
     fn insert_directives(&mut self, config: &mut Config) {
         let mut directives = Vec::with_capacity(config.graph.directives.len());
-        for directive in take(&mut config.graph.directives) {
+        for directive in &config.graph.directives {
```

Comment:
why not `take`?

Reply:
Since `insert_directives()` is called before we insert the object fields, we ended up not being able to know which object fields were inaccessible because the directives vec was empty. But now I grouped the inaccessible checks in `initialize()`, so it's likely we can revert this line back to `take()`, I'll try.